src/solvers/base_solver.cpp

```diff
@@ -189,10 +188,8 @@ void BaseSolver::OutputState() const
 
     case OutputType::GLVis: {
       for (auto &state : m_state) {
-        std::ostringstream sol_name;
-        sol_name << m_root_name << "-" << state.name << "." << std::setfill('0') << std::setw(6) << m_cycle << "."
-                 << std::setfill('0') << std::setw(6) << m_rank - 1;
-        std::ofstream osol(sol_name.str().c_str());
+        std::string   sol_name = fmt::format("{0}-{1}.{2:0>6}.{3:0>6}", m_root_name, state.name, m_cycle, m_rank);
```

Comment:
`m_rank-1` became `m_rank`, not sure if intended or not

Reply:
Yep, that was intended. I'm not sure how this worked at all before... 🤷‍♀️